File: src/Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(n == 0 || n == 1)
            return n;
        int ret = 0;
        int pre = 0;
        int old = A[0];
        for(int i=1; i<n; i++)
        {
            if(old != A[i])
            {
                ret++;
                old = A[i];
                swap(A[++pre], A[i]);
            }
        }
        return ret+1;
    }
};

File: python/Remove Duplicates from Sorted Array.py
class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        if len(A) == 0 or len(A) == 1:
            return len(A)

        ret = 0
        old = A[0]
        pre = 0
        for i in range(1, len(A)):
            if A[i] != old:
                old = A[i]
                pre += 1
                tmp = A[pre]
                A[pre] = A[i]
                A[i] = tmp
                ret += 1

        return ret + 1 
